sentry-kotlin-multiplatform-gradle-plugin/src/main/java/io/sentry/kotlin/multiplatform/gradle/ManualFrameworkPathSearchValueSource.kt

```diff
@@ -0,0 +1,61 @@
+package io.sentry.kotlin.multiplatform.gradle
+
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.ValueSource
+import org.gradle.api.provider.ValueSourceParameters
+import org.gradle.process.ExecOperations
+import java.io.ByteArrayOutputStream
+
+abstract class ManualFrameworkPathSearchValueSource :
+    ValueSource<String?, ManualFrameworkPathSearchValueSource.Parameters> {
+    interface Parameters : ValueSourceParameters {
+        val frameworkType: Property<FrameworkType>
+        val basePathToSearch: Property<String>
+    }
+
+    @get:javax.inject.Inject
+    abstract val execOperations: ExecOperations
+
+    override fun obtain(): String? {
+        val frameworkType = parameters.frameworkType.get()
+        val basePathToSearch =
+            parameters.basePathToSearch.orNull
+                ?: "\"${System.getProperty("user.home")}/Library/Developer/Xcode/DerivedData\""
+        return findFrameworkWithFindCommand(frameworkType, basePathToSearch)
+    }
+
+    /**
+     * Returns valid framework if exists with the find command.
+     * If the command finds multiple frameworks, it will return the one who has been modified or used most recently.
+     */
+    private fun findFrameworkWithFindCommand(
+        frameworkType: FrameworkType,
+        basePathToSearch: String
+    ): String? {
+        val output = ByteArrayOutputStream()
+
+        val xcFrameworkName =
+            if (frameworkType == FrameworkType.STATIC) "Sentry.xcframework" else "Sentry-Dynamic.xcframework"
+        execOperations.exec {
+            it.commandLine(
+                "bash",
+                "-c",
+                "find $basePathToSearch " +
+                    "-name $xcFrameworkName " +
+                    "-exec stat -f \"%m %N\" {} \\; | " +
+                    "sort -nr | " +
+                    "cut -d' ' -f2-"
+            )
+            it.standardOutput = output
+            it.isIgnoreExitValue = true
+        }
+
+        val stringOutput = output.toString("UTF-8")
+        return if (stringOutput.lineSequence().firstOrNull().isNullOrEmpty()) {
```

Comment:
I think we should also account for `exitValue != 0`, if e.g. the find command fails/doesn't exist/etc. You can check how we do it here (and also could pipe the error out stream to later print it out for easier debugging): https://github.com/getsentry/sentry-android-gradle-plugin/blob/14a69bcdc41d00f01fa8fdd7fa7648715da89109/plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt#L484-L527

Reply:
it's also because you're just getting the first line in the output here, but it could be `Failed to execute 'find' ...` or something like that, and we don't do any validation over it as far as I can see